flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/debezium/reader/BinlogSplitReader.java

```diff
@@ -257,6 +259,10 @@ private boolean shouldEmit(SourceRecord sourceRecord) {
 
             // only the table who captured snapshot splits need to filter
             if (finishedSplitsInfo.containsKey(tableId)) {
+                // if backfill skipped, don't need to filter
+                if (isBackfillSkipped) {
+                    return true;
```

Comment:
Currently, there are relatively few tests on Skip backfill. Can you add necessary tests to verify its correctness？

Reply:
Add test:
org.apache.flink.cdc.connectors.mysql.debezium.reader.BinlogSplitReaderTest#testSnapshotScanSkipBackfillWithPostHighWatermark
org.apache.flink.cdc.connectors.mysql.debezium.reader.BinlogSplitReaderTest#testSnapshotScanSkipBackfillWithPreHighWatermark